packages/velo-external-db-core/lib/converters/transform.js

```diff
@@ -54,6 +55,22 @@ const defaultValueFor = (f) => {
 const prepareForInsert = (item, fields) => fields.reduce((pv, f) => ({ ...pv, [f.field]: item[f.field] || defaultValueFor(f) }), {})
 const prepareForUpdate = (item, fields) => fields.reduce((pv, f) => f.field in item ? ({ ...pv, [f.field]: item[f.field] }) : pv, {})
 
+
+const createProjectionArr = (fields, projection) => {
+    const fieldsArr = fields.map(f => f.field)
+ 
+    if (isEmptyProjection(projection)) return fieldsArr
```

Comment:
can we have an empty array here ? why ? is there a case where a table doesn't have any fields ?

Reply:
Same here as below, I treated the projection as input from outside, and not as all the fields

Reply:
so right now the logic is, get al fields and pass it to the next layer, in the future we will handle the customer input.
in case the customer doesn't add any fields, we will need to decide how to behave, i will prefer to reject the request rather than trying to execute it.